### ISSUE TITLE
compile error when converting selectmodule to AC due to missing #define-s

### ISSUE THREAD

**Issue by taleinat** [python/cpython#76107]
BPO | [31926](https://bugs.python.org/issue31926)
--- | :---
Nosy | @vstinner, @taleinat, @larryhastings, @serhiy-storchaka
PRs | <li>python/cpython#4230</li><li>python/cpython#4253</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-11-03.13:46:37.042>
created_at = <Date 2017-11-02.15:13:55.394>
labels = ['build', '3.7', 'expert-argument-clinic']
title = 'compile error when converting selectmodule to AC due to missing #define-s'
updated_at = <Date 2017-11-03.13:46:37.041>
user = 'https://github.com/taleinat'
```

bugs.python.org fields:
```python
activity = <Date 2017-11-03.13:46:37.041>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-11-03.13:46:37.042>
closer = 'vstinner'
components = ['Argument Clinic']
creation = <Date 2017-11-02.15:13:55.394>
creator = 'taleinat'
dependencies = []
files = []
hgrepos = []
issue_num = 31926
keywords = ['patch']
message_count = 5.0
messages = ['305427', '305428', '305464', '305481', '305482']
nosy_count = 4.0
nosy_names = ['vstinner', 'taleinat', 'larry', 'serhiy.storchaka']
pr_nums = ['4230', '4253']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'compile error'
url = 'https://bugs.python.org/issue31926'
versions = ['Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by taleinat** [comment]
I'm converting [Modules/selectmodule.c.h](https://github.com/python/cpython/blob/main/Modules/selectmodule.c.h) as part of issue bpo-20182.

After converting the code and running [Tools/clinic/clinic.py](https://github.com/python/cpython/blob/main/Tools/clinic/clinic.py), compilation is failing on Windows due to the following block \*not* being generated by in the [Modules/clinic/selectmodule.c.h](https://github.com/python/cpython/blob/main/Modules/clinic/selectmodule.c.h) file:

```c
#ifndef SELECT_POLL_METHODDEF
    #define SELECT_POLL_METHODDEF
#endif /* !defined(SELECT_POLL_METHODDEF) */
```


Adding the block manually fixes the issue.

This seems to be caused by a bug in clinic.py. Attaching PR fix a fix immediately.

**Comment by taleinat** [comment]
See [PR 4230](https://github.com/python/cpython/pull/4230).

**Comment by serhiy-storchaka** [comment]
New changeset 4f57409a2f7bdf8fb559cddc7c6533ca2c471c67 by Serhiy Storchaka (Tal Einat) in branch 'master':
bpo-31926: fix missing \*_METHODDEF statements by argument clinic (bpo-4230)
https://github.com/python/cpython/commit/4f57409a2f7bdf8fb559cddc7c6533ca2c471c67

**Comment by vstinner** [comment]
New changeset f8b3f6b178e48773cd7298141cbaf408c6917e41 by Victor Stinner (Tal Einat) in branch '3.6':
[3.6] bpo-31926: fix missing \*_METHODDEF statements by argument clinic (GH-4230) (bpo-4253)
https://github.com/python/cpython/commit/f8b3f6b178e48773cd7298141cbaf408c6917e41

**Comment by vstinner** [comment]
Thank you Tal Einat for your bug report and your fix. It's now merged into 3.6 and master (future 3.7).
